File: Solutions/0392-is-subsequence/solution.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int l=0,r=0;
        
        while(l<s.length() && r<t.length()) {
            if(s[l]==t[r]) {
                l++, r++;
            } else {
                r++;
            }
        }
        
        if(l==s.length()) return true;
        return false;
        
    }
};

File: Solutions/0392-is-subsequence/solution.go
func isSubsequence(s string, t string) bool {
    
        l:=0
        r:=0
        
    for l<len(s) && r<len(t) {
        if s[l]==t[r] {
            l++
            r++
        } else {
            r++
        }
    }
    
    if l==len(s) {
        return true
    }
    
    return false
}
